### ISSUE TITLE
Herbstluftwm and lightdm

### ISSUE THREAD

**Issue by arcolinuxz** [herbstluftwm/herbstluftwm#599]
ArcoLinux uses lightdm as our standard display manager on 15 desktops.

We were wondering if you can help with the strange "look" when we start up herbstluftwm and how to overcome it.

**After login we see a black screen with a red border.** 

Can we get rid of that somehow? 
We do want to keep lightdm as display manager.

Check video at 40:46
https://www.youtube.com/watch?v=RzA8zgbq-nE

**Comment by bazeeel** [comment]
yes i have that too with lightdm but ...if i start the herbstluftwm inside .xinitrc  then no more that strange black frame/window with red borders.

**Comment by arcolinuxz** [comment]
Could you please assist?

**Comment by bl00dy1837** [comment]
I don't know lightdm, I use .xinitrc.

For testing press your Mod Key (default ALT) + Return. Then should xterm start.

Read: https://herbstluftwm.org/tutorial.html or a tutorial as video https://www.youtube.com/watch?v=81fL98ZN7_g .

**Comment by bazeeel** [comment]
> Could you please assist?

if not exist just create .xinitrc 
file
inside write exec herbsluftwm --locked

**Comment by arcolinuxz** [comment]
> For testing press your Mod Key (default ALT) + Return. Then should xterm start.

Why do I need to start up xterm?
Do do what?

**Comment by bl00dy1837** [comment]
> > For testing press your Mod Key (default ALT) + Return. Then should xterm start.
> 
> Why do I need to start up xterm?
> To do what?

Only to test if herbstluftwm works. After starting herbstluftwm default config there is only a black screen.

herbstluftwm is a tilling manager. there is no desktop/application menu. You need to define it by yourselfe.

I use as topbar "demu2" and to start applications "rofi".

**Comment by tiosgz** [comment]
Sorry for responding so late… is this still actual for you?

The black screen is the main frame before you unlock HLWM: https://github.com/herbstluftwm/herbstluftwm/blob/78be1fa7d934fb700ce576f3c828aa82f71241cb/share/autostart#L157

You could maybe reduce the time partly by rearranging the autostart so that only visual changes happen before unlocking, & by `chain`ing the commands (but that may reduce readability if you aren’t careful — which is something you care about, do I remember correctly?)

I don’t recommend unlocking immediately (or not locking at all), coz, well, it usually reduces performance (see `lock`/`unlock` in the manpage).

**Comment by Barbarossa93** [comment]
I can confirm that adding `--locked` when starting `herbstluftwm` removes the black screen with red border. I think adding it to the `xsession` file should be the default

**Comment by erikdubois** [comment]
Thanks for merging that fix into your desktop file.
In this video you can see the effect of adding --locked to the .desktop file.

https://youtu.be/rz_sRWis6cs

**Comment by NNBnh** [comment]
Hi, I'm a AwesomeWM and I'm trying to switch to HLWM but encounter the same 
"black window with red border" problem...

> if not exist just create .xinitrc file inside write
> ```sh
> #!/usr/bin/env sh
> exec herbsluftwm --locked
> ```
> start with startx in tty

I'm already using TTY and `startx` as a display manager replacement so this is great, but as soon as `herbstclient unlock` is executed the black window with red border appear and it won't disappear no matter what I do (`xkill`, `herbstclient lock`, ...)

This must be a bug right?

**Comment by t-wissmann** [comment]
The black window with the red border is a "frame" in hlwm's language. The fact that it looks so wrong is due to the fact that your `autostart` does not seem to be executed (it configures the frames' appearance to something more pleasant). Have you copied the default autostart to `~/.config/herbstluftwm/autostart` and is it executable?
